independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverInfo.java

```diff
@@ -99,8 +99,9 @@ static ObserverInfo create(String id, BeanDeployment beanDeployment, DotName bea
             } else {
                 info = beanClass.toString();
             }
-            LOGGER.warnf("The observer %s makes use of %s transactional observers but no " +
-                    "JTA capabilities were detected.", info, transactionPhase);
+            LOGGER.warnf(
```

Comment:
From the chat I understood we wanted to only show this when `!jtaCapabilities` and avoid it entirely in case of no tx in progress? E.g. IMO the condition on L#94 should only check for JTA cappabilities.

Reply:
According to the spec https://jakarta.ee/specifications/cdi/2.0/cdi-spec-2.0.html#transactional_observer_methods
```
A transactional observer method may be declared by specifying any value other than IN_PROGRESS for during
```

Removing the check for `!transactionPhase.equals(TransactionPhase.IN_PROGRESS)` would trigger this warning for normal observers as well when no jta is present.

Reply:
Sorry I wasn't clear - I meant checking if the method has *any* transaction phase and there are no JTA capabilities.

However, looking into code it looks like we are automatically giving all OMs a default phase (see [this code](https://github.com/quarkusio/quarkus/blob/master/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ObserverInfo.java#L274-L283)). @mkouba do you happen to know why is that?

Reply:
We use `TransactionPhase.IN_PROGRESS` if no explicit `Observes.during()` is set, i.e. we use the default value from the annotation. Note that Jandex does not return the default value unless `AnnotationInstance.valueWithDefault()` is used (because defaults are stored on the annotation's defining class).